FlaskApp/app/views.py

```diff
@@ -5,41 +5,82 @@
 # dude these schema things should not be in the view.. this basically means
 # everytime when you refresh the page you have to reconnect the server??
 # I believe this should be in init file and the you shoudl find a way to maintain through model.
+from datetime import datetime, timedelta
 from flask import render_template, redirect, url_for, request
 from app.schema.tweets import Tweets_sche
 import pymongo
 from pymongo import MongoClient
 from pymongo.errors import ConnectionFailure
-# this file should works as part of controllers for routes
-# if we have more actions later we can have a folder for controllers later but now lets keep it here.
-# all routes should go here
+import logging
+from logging import Formatter, FileHandler
+
+#Setup the logger
+LOGGER = logging.getLogger('streamer_logger')
+file_handler = FileHandler('streamer.log')
+handler = logging.StreamHandler()
+file_handler.setFormatter(Formatter(
+        '%(thread)d %(asctime)s %(levelname)s: %(message)s '
+        '[in %(pathname)s:%(lineno)d]'
+))
+handler.setFormatter(Formatter(
+        '%(thread)d %(asctime)s %(levelname)s: %(message)s '
+        '[in %(pathname)s:%(lineno)d]'
+))
+LOGGER.addHandler(file_handler)
+LOGGER.addHandler(handler)
+LOGGER.setLevel(logging.DEBUG)
+
 @app.route("/")
 def index():
+    return render_template('index.html')
 
-	#----DB INIT BEGIN---------------------------------------------
-
-	MONGO_HOST = 'localhost'
-	MONGO_PORT = 27017
 
-	#Try to connect to MongoDB
-	try:
-		client = MongoClient(MONGO_HOST, MONGO_PORT)
-		db = client.tinyjumbo
-		tweet_collection = db.google
-		print "Sucees connect to DB"
+@app.route("/barchart")
+def bar():
+    #----DB INIT BEGIN---------------------------------------------
 
-	except ConnectionFailure:
-		LOGGER.error('Could not connect to MongoDB, aborting Flask app...')
-		sys.exit(-1)
-	#----DB INIT END---------------------------------------------
+    MONGO_HOST = '162.243.122.37'
+    MONGO_PORT = 27017
 
+    #Try to connect to MongoDB
+    try:
+        client = MongoClient(MONGO_HOST, MONGO_PORT)
+        db = client.tinyjumbo
+        tweet_collection = db.tcount
+        LOGGER.info('connecting to DB...')
+        print "Successfully connect to DB ---"
 
-	#read from DB and store it to Tweets_sche object
-	db_read = tweet_collection.find_one()
-	tweet= Tweets_sche(db_read["time"],db_read["text"],db_read["hashtages"])
+    except ConnectionFailure:
+        LOGGER.error('Could not connect to MongoDB, aborting Flask app...')
+        sys.exit(-1)
+	#----DB INIT END---------------------------------------------
 
 
-	return render_template('index.html',info=tweet)
+    #read from DB and store it
+    print "start to get collections"
+    data_set = []
+    date_set = []
+    count_set = []
+    print "create an empty set"
+    now = datetime.now()
+    start = datetime(2016, 2, 18, 20, 01, 01)
```

Comment:
I would suggest define a method for this. and then call it. you may need this function muti times.